Sources/Extensions/SwiftStdlib/DictionaryExtensions.swift

```diff
@@ -145,7 +145,49 @@ public extension Dictionary where Key: StringProtocol {
             }
         }
     }
+}
+
+// MARK: - Subscripts
+public extension Dictionary {
 
+    /// SwifterSwift: Deep fetch or set a value from nested dictionaries.
+    ///
+    ///        var dict =  ["key": ["key1": ["key2": "value"]]]
+    ///        dict[path: ["key", "key1", "key2"]] = "newValue"
+    ///        dict[path: ["key", "key1", "key2"]] -> "newValue"
+    ///
+    /// - Note: Value fetching is iterative, while setting is recursive.
+    ///
+    /// - Complexity: O(N), _N_ being the length of the path passed in.
+    ///
+    /// - Parameter path: An array of keys to the desired value.
+    ///
+    /// - Returns: The value for the key-path passed in. `nil` if no value is found.
+    public subscript(path path: [Key]) -> Any? {
+        get {
+            guard !path.isEmpty else { return nil }
+            var result: Any? = self
+            for key in path {
+                if let element = (result as? [Key: Any])?[key] {
+                    result = element
+                } else {
+                    return nil
+                }
+            }
+            return result
+        }
+        set {
+            if let first = path.first {
+                if path.count == 1, let new = newValue as? Value {
```

Comment:
Is this cast necessary?

Reply:
Yes. If I remove it, I get `Cannot assign value of type '[Dictionary<Key, Value>.Key : Any]' (aka 'Dictionary<Key, Any>') to type 'Value?'`